src/python/pants/backend/python/tasks/wrapped_pex.py

```diff
@@ -4,57 +4,68 @@
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 
+import logging
 from builtins import object
 from copy import copy
 
 
+logger = logging.getLogger(__name__)
+
+
 class WrappedPEX(object):
   """Wrapper around a PEX that exposes only its run() method.
 
   Allows us to set the PEX_PATH in the environment when running.
   """
 
   _PEX_PATH_ENV_VAR_NAME = 'PEX_PATH'
+  _PEX_PYTHON_PATH_ENV_VAR_NAME = 'PEX_PYTHON_PATH'
 
-  # TODO(benjy): I think we can get rid of the interpreter argument.
-  # In all cases it appears to be set to pex.interpreter.
-  def __init__(self, pex, interpreter, extra_pex_paths=None):
+  def __init__(self, pex, extra_pex_paths=None):
     """
     :param pex: The main pex we wrap.
-    :param interpreter: The interpreter the main pex will run on.
     :param extra_pex_paths: Other pexes, to "merge" in via the PEX_PATH mechanism.
     """
     self._pex = pex
-    self._interpreter = interpreter
     self._extra_pex_paths = extra_pex_paths
 
   @property
   def interpreter(self):
-    return self._interpreter
+    return self._pex._interpreter
 
   def path(self):
     return self._pex.path()
 
   def cmdline(self, args=()):
     cmdline = ' '.join(self._pex.cmdline(args))
+
+    def render_env_var(key, value):
+      return '{key}={value}'.format(key=key, value=value)
+
+    env_vars = [(self._PEX_PYTHON_PATH_ENV_VAR_NAME, self._pex._interpreter.binary)]
+
     pex_path = self._pex_path()
     if pex_path:
-      return '{env_var_name}={pex_path} {cmdline}'.format(env_var_name=self._PEX_PATH_ENV_VAR_NAME,
-                                                          pex_path=pex_path,
-                                                          cmdline=cmdline)
-    else:
-      return cmdline
+      env_vars.append((self._PEX_PATH_ENV_VAR_NAME, pex_path))
+
+    return '{execution_control_env_vars} {cmdline}'.format(
+      execution_control_env_vars=' '.join(render_env_var(k, v) for k, v in env_vars),
+      cmdline=cmdline
+    )
 
   def run(self, *args, **kwargs):
+    env = copy(kwargs.pop('env', {}))
+
+    # Hack around bug in PEX where custom interpreters are not forwarded to PEXEnvironments.
+    # TODO(John Sirois): Remove when XXX is fixed.
```

Comment:
s/`XXX`/a ticket?

Reply:
Yup - just finished filing on the pex side. I'll send up a final comment-only diff with issue inks here as soon as this goes green then merge: https://github.com/pantsbuild/pex/issues/522

Reply:
Fixed.